code/src/main/kotlin/com/expediagroup/sdk/v2/core/apache/util/HttpClientUtil.kt

```diff
@@ -21,32 +25,58 @@ fun createHttpClient(configuration: ClientConfiguration): HttpClient {
         .setDefaultRequestConfig(createRequestConfig(configuration))
         .setMaxConnTotal((configuration as MaxConnectionsTotalTrait).getMaxConnectionsTotal())
         .setMaxConnPerRoute((configuration as MaxConnectionsPerRouteTrait).getMaxConnectionsPerRoute())
-        .addInterceptorLast(ExpediaGroupRequestInterceptor())
-        .addInterceptorLast(ExpediaGroupResponseInterceptor())
         .build()
 }
 
+/**
+ * Retrieves a singleton instance of HttpClient configured with the provided ClientConfiguration.
+ *
+ * @param configuration The configuration object implementing ClientConfiguration, containing the necessary settings for HttpClient.
+ * @return A singleton instance of HttpClient configured according to the provided ClientConfiguration.
+ */
 fun getSingletonHttpClient(configuration: ClientConfiguration) =
     CreateSingletonHttpClientLambda.execute(configuration)
 
 
+/**
+ * Singleton pattern implementation to create and manage a single instance of HttpClient.
+ *
+ * This class inherits from a function type `(ClientConfiguration) -> HttpClient`
+ * and allows the creation of a singleton HttpClient instance based on the provided configuration.
+ *
+ * The companion object maintains the singleton instance and provides methods for accessing it.
+ */
```

Comment:
Kind of similar notes on this (regarding the singleton implementation approach).

Reply:
Done.